_posts/2020-04-11-native-mobile-to-flutter.markdown

```diff
@@ -0,0 +1,114 @@
+- The application polish and user experience is on par with our native iOS application and easily superior to our native android application.
+- Most importantly, we experienced no Flutter related bugs or unpredictable behaviors after the launch.
+
+### How long did this take? 
+
+- Project spanned Aug 2019 - Dec 2019 
+- Initial release to Android users in February 2020.
+- Release to all users Android + iOS March 2020.
+
+### What problems were we hoping to solve?
+
+We had two working native mobile applications, but lacked feature parity for Android and iOS users.
+
+Android and iOS development took place at different paces, with varying quality and cost.
+
+Maintaining redundancy across two mobile teams would have required us to maintain four to six developers rather than as few as two to three in a cross platform environment.
+
+_Our key objectives were..._
+
+- feature parity on both platforms.
+- a unified mobile team with higher quality peer reviews. 
+- increased velocity and reduced cost per feature. 
+- higher quality software. 
+- scalability and redundancy of our mobile team.
+
+This finally seemed possible as the cross-platform mobile space dramatically matured over the last two years: there were two great options to choose from, _Flutter_ and _React Native_.
+
+
+### Why did we end up choosing Flutter? 
+
+The biggest challenge we faced was the timeline within which we had to complete the switch: just 5 short months.  Our native apps had 140,000 lines of code between the two platforms, so we needed to factor in both the learning curve and development speed.
+
+_Regardless of our choice our entire team would have to..._
+
+1. learn a new programming language and UI framework
+3. add sufficient test coverage to confidently launch the application
+4. deliver a highly secure application. 
+
+At the end of the day, the developers called it.  Flutter felt much more approachable and less like web a web platform than React Native.
+
+#### Things we liked about React:
+
+- Maturity and community size: React Native has been in production for much longer.
+- Testing utilities are definitely better on React.
+- More third party library support.
+
+#### Things we liked about Flutter:
+
+- Without javascript developers on our mobile team, the learning curve for Dart/Flutter was substantially lower than Javascript/React Native. 
+- Flutter offered clean, straight-forward conventions that were easy to learn, expressive and explicit.
+- Android Studio offered a turnkey development environment with performance monitoring tools, hot reload and step through debugging and inspection.  Since most of our developers are Android developers, the IDE was an added bonus due to its familiarity.
+- Flutter does not rely on any native UI components, so it's largely immune to breaking changes introduced by Apple, Google, or Android device manufacturers.
```

Comment:
Wondering if you'd like to mention the the stateful hot reload as a productivity-boost for Flutter?

Reply:
That's a good one ... also way fewer platform specific bugs on android due to less reliance on the Android operating system.